pkgs/development/libraries/boost/generic.nix

```diff
@@ -114,6 +114,10 @@ stdenv.mkDerivation {
     sha256 = "sha256-KlmIbixcds6GyKYt1fx5BxDIrU7msrgDdYo9Va/KJR4=";
   });
 
+  postPatch = lib.optional (enablePython && (version == "1.79.0" || version == "1.80.0")) ''
+    sed -i '/Py_TPFLAGS_HAVE_GC/d' libs/python/src/object/enum.cpp
+  '';
+
```

Comment:
Can we fetch the patch and apply it for better self-documentation? I tested the following, which seems to work:
```
  ++ lib.optionals (lib.versionOlder version "1.81") [
    (fetchpatch {
      name = "python311-compatibility.patch";
      url = "https://github.com/boostorg/python/commit/a218babc8daee904a83f550fb66e5cb3f1cb3013.patch";
      hash = "sha256-IHxLtJBx0xSy7QEr8FbCPofsjcPuSYzgtPwDlx1JM+4=";
      stripLen = 1;
      extraPrefix = "libs/python/";
    })
  ];
```
1. I tested that this applies cleanly to `boost175`, `boost177`, and `boost178` (and that `boost175` builds). Since `boost179` is the default, I expect that applying the patch to these three too will not trigger too many rebuilds.
2. Since the number of rebuilds is already very high, perhaps we can just apply the patch regardless of whether `enablePython` is requested for simplicity.

Reply:
thanks a lot for your review. I'm pretty sure I tried with those exact stripLen and extraPrefix, but couldn't get them to work, and in the end I got distracted. 
But I'll try again :D

Reply:
Done, I don't know why my previous tests were wrong, but I can confirm this is working now. I added you as co-autho on the commit :)